src/bin/add/args.rs

```diff
@@ -106,6 +112,21 @@ impl Args {
 
         Ok(vec![dependency])
     }
+
+    fn get_upgrade_prefix(&self) -> Option<&'static str> {
+		self.flag_upgrade.clone().map(|flag|
```

Comment:
Sorry, I think I told you to use `.map`, right? That was not the best option. My bad.

`.map` does this: `let new_option = match old_option { Some(val) => Some(mapper(val)), None => None };`.

Right now, you are taking an `Option<String>` and, if it's `Some(String)`, fill it with an `Option<String>` (you `map` return an `Option`). So, you end up with an `Option<Option<String>>` and call `unwrap_or` on it, to get that `Option`out of the `Option`.

Pretty confusing, right? Luckily, `Option` also has `.and_then`, which does exactly what we want here: Apply a function that can fail (i.e., returns on `Option`) if it's `Some(val)`.

So `.and_then` is basically this: `let new_option = match old_option { Some(val) => mapper(val), None => None };` (notice the missing `Some(…)` around the `mapper` call).

Long story short, if you use `.and_then` instead of `.map` you can get rid of the `unwrap_or` 😄